backend/main.py

```diff
@@ -37,46 +42,134 @@
 setup_logging(level="INFO")
 logger = get_logger(__name__)
 
-app = FastAPI()
 
-origins = ["*"]  # Update this with the frontend domain in production
+def get_incluster_namespace() -> str:
+    """Get the current Kubernetes namespace."""
+    try:
+        with open("/var/run/secrets/kubernetes.io/serviceaccount/namespace") as file:
+            return file.read().strip()
+    except Exception:
+        return "default"
+
+
+def wait_for_service_ready(
+    service_name: str,
+    namespace: str,
+    timeout_seconds: int = 300,
+    interval_seconds: int = 5,
+) -> bool:
+    """Wait for a Kubernetes service to be ready."""
+    start_time = time.time()
+
+    while time.time() - start_time < timeout_seconds:
+        try:
+            config.load_incluster_config()
+            core_v1 = client.CoreV1Api()
+            endpoints = core_v1.read_namespaced_endpoints(
+                name=service_name, namespace=namespace
+            )
+
+            if endpoints.subsets:
+                for subset in endpoints.subsets:
+                    if subset.addresses:
+                        logger.info(
+                            f"Service '{service_name}' in namespace \
+                                '{namespace}' is ready."
+                        )
+                        return True
+
+        except client.ApiException as e:
+            if e.status != 404:  # Ignore 404 if service not yet created
+                logger.error(f"Error checking endpoints: {e}")
+
+        logger.info(
+            f"Waiting for service '{service_name}' in namespace \
+                '{namespace}' to be ready..."
+        )
+        time.sleep(interval_seconds)
+
+    logger.warning(
+        f"Timeout waiting for service '{service_name}' in namespace '{namespace}'."
+    )
+    return False
+
+
+async def sync_all_services():
```

Comment:
I think these sync calls are no longer needed as mcp servers, models and knowledge bases will always be fetched from llama stack. This can be removed later. May be we should log an issue or write a jira for the task.

Reply:
your call, i can certainly remove them now if others agree

Reply:
I think a separate PR would be better even though its minor as I think we also need to remove some database stuff.